client/src/controllers/RevealController.ts

```diff
@@ -219,6 +228,35 @@ export class RevealController extends Controller<HTMLElement> {
       });
   }
 
+  setSavedClosedValue(closed) {
+    const id = this.element.id;
+
+    if (!this.saveStateValue || !id) return;
+
+    try {
+      localStorage.setItem(
+        `wagtail:reveal-${id}-closed`,
+        closed ? 'true' : 'false',
+      );
+    } catch {
+      // Skip saving the preference if localStorage isn’t available.
+    }
+  }
+
+  getSavedClosedValue(defaultClosed) {
+    const id = this.element.id;
+
+    if (!this.saveStateValue || !id) return defaultClosed;
+
+    let saved;
+    try {
+      saved = localStorage.getItem(`wagtail:reveal-${id}-closed`);
```

Comment:
```suggestion
      saved = localStorage.getItem(`wagtail:${this.identifier}-${id}-closed`);
```

As above, while id will of course be unique, I find it a useful pattern for the controller code to be ignorant to its identifier.